broadinstitute/single_cell_portal_core
# Preserve color assignments during cell filtering (SCP-5310)

#### BACKGROUND
When using the cell filtering feature, if a user deselects any label that results in an entire group being removed from the plot, all color assignments will shift up by one value.  This in essence changes the meaning of the plot, as the color assignments do not persist.  This is confusing and visually jarring to the end user.

#### CHANGES
Now, original color assignments are persisted when using the cell filtering feature.  Any completely filtered groups will move to the `--Filtered--` trace, but the colors will not shift in accordance.

Existing bug:

https://github.com/broadinstitute/single_cell_portal_core/assets/729968/804d5309-f784-4761-b63c-f2e9a3fae1ae

Fixed UX:

https://github.com/broadinstitute/single_cell_portal_core/assets/729968/cc889cea-0945-4574-9293-056baf9e20d8

#### MANUAL TESTING
1. Boot as normal, sign in and load the `Human milk - differential expression` study
2. Once enabled, click the `Cell filtering` button for the `All Cells UMAP` clustering
3. Deselect `macrophage` from the `cell_type__ontology_label` annotation
4. Confirm that the color for `LC2` remains grey, as pictured above
